### PR TITLE
More details on API errors

### PR DESCRIPTION
**Pull request by MihaiBojin**
Errors are not augmented with the response body
```
Error: failed to execute GET request to http://ec2-52-213-250-143.eu-west-1.compute.amazonaws.com:9080/api/public/v1.0/groups
status code: 403
response: 403 Forbidden
details: map[detail:IP address 37.228.253.237 is not allowed to access this resource. error:%!!(MISSING)s(float64=403) errorCode:IP_ADDRESS_NOT_ON_WHITELIST parameters:[37.228.253.237] reason:Forbidden]

{
	"links": null,
	"results": null,
	"totalCount": 0
}
```

### CODE REVIEW

File: pkg/httpclient/client.go [hunk 1]
```diff
@@ -124,27 +124,27 @@ func NewClientWithAuthentication(username string, password string) BasicHTTPOper
...
 // GetJSON retrieves the specified URL
 func (cl basicHTTPClient) GetJSON(url string) HTTPResponse {
-	return cl.genericJSONRequest("GET", url, nil, []int{http.StatusOK})
+	return cl.genericJSONRequest(http.MethodGet, url, nil, []int{http.StatusOK})
 }
 
 // PostJson executes a POST request, sending the specified body, encoded as JSON, to the passed URL
 func (cl basicHTTPClient) PostJSON(url string, body io.Reader) HTTPResponse {
-	return cl.genericJSONRequest("POST", url, body, []int{http.StatusOK})
+	return cl.genericJSONRequest(http.MethodPost, url, body, []int{http.StatusOK})
```
Comment (gssbzn) [inline pkg/httpclient/client.go]: Can you add `http.StatusCreated` to the expected status? I had to add this locally to get the create project to work

**Review by gssbzn — APPROVED** [review]
LGTM

File: pkg/httpclient/client.go (hunk 1 above)
Comment (MihaiBojin) [reply pkg/httpclient/client.go]: yes

Comment (MihaiBojin) [reply pkg/httpclient/client.go]: @gssbzn refactored.

File: go.mod [hunk 2]
```diff
@@ -4,5 +4,10 @@ go 1.12
 
 require (
 	github.com/Sectorbob/mlab-ns2 v0.0.0-20171030222938-d3aa0c295a8a
+	github.com/mitchellh/go-homedir v1.1.0
```
Comment (MihaiBojin) [inline go.mod:7]: @gssbzn looks like the dep changes were not committed in your base branch

Comment (gssbzn) [reply go.mod:7]: I haven't merged so no idea how they got into this PR, also I have no idea why they are not in my pr 🤷‍♂

**Review by gssbzn — APPROVED** [review]
LGTM

File: pkg/opsmanager/api_create_agent_api_key.go [hunk 3]
```diff
@@ -4,6 +4,6 @@ import "errors"
 
 // CreateAgentAPIKEY
 // https://docs.opsmanager.mongodb.com/master/reference/api/agentapikeys/create-one-agent-api-key/
-func (api opsManagerClient) CreateAgentAPIKEY(projectID string, name string) (interface{}, error) {
+func (client opsManagerClient) CreateAgentAPIKEY(projectID string, name string) (interface{}, error) {
```
Comment (gssbzn) [inline pkg/opsmanager/api_create_agent_api_key.go:7]: `go` prefers single letters here, `c` but is just a recommendation

**Pull request merged**